src/dotty/tools/dotc/core/pickling/TreePickler.scala

```diff
@@ -531,8 +535,16 @@ class TreePickler(pickler: TastyPickler) {
       withLength { pickleType(ann.symbol.typeRef); pickleTree(ann.tree) }
     }
 
+    def updateMapWithDeltas[T](mp: collection.mutable.Map[T, Addr]) = {
+      mp.map{
+        case (key, addr) => (key, adjusted(addr))
```

Comment:
I'd use a simple loop here:

    for ((key, addr) <- mp) mp(key) = adjusted(addr)

Shorter, clearer, faster.